### PR TITLE
Add function to calculate the `crc32` from Compressed Commitment data

### PR DESCRIPTION
**Pull request by the-real-adammork**
Soundtrack of this PR: [Duke Hugh - 2017 Heatwave](https://www.youtube.com/watch?v=-MB4UzSsHmM&list=RD__8L_E9ije0&index=2)

### Motivation

The iOS SDK needs a way to calculate the crc32 checksum from Compressed Commitment data

### In this PR
* New function `mc_tx_out_commitment_crc32` in transaction.rs

### Future Work
* Unit tests for functions in /libmobilecoin

### CODE REVIEW

**Comment by garbageslam** [comment]
how come there are 2 PRs, should there just be one? https://github.com/mobilecoinfoundation/mobilecoin/pull/1454

**Comment by the-real-adammork** [comment]
> how come there are 2 PRs, should there just be one? #1454

closed the other one

**Review by jcape — APPROVED** [review]

**Review by samdealy — APPROVED** [review]

File: libmobilecoin/Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ aes-gcm = "0.9.4"
 displaydoc = "0.2"
 libc = "0.2"
 protobuf = "2.22.1"
+crc = "1.8.1"
```
Comment (remoun) [inline libmobilecoin/Cargo.toml:17]: Why 1.8.1 and not the latest 2.1.0?

File: libmobilecoin/include/transaction.h [hunk 2]
```diff
@@ -52,6 +52,20 @@ bool mc_tx_out_matches_any_subaddress(
...
 MC_ATTRIBUTE_NONNULL(1, 2, 3, 4);
 
+/// # Preconditions
+///
+/// * `tx_out_commitment` - must be a valid CompressedCommitment
+///
+/// # Errors
+///
+/// * `LibMcError::InvalidInput`
+bool mc_tx_out_commitment_crc32(
```
Comment (remoun) [inline libmobilecoin/include/transaction.h:62]: Can we have a basic test that exercises these methods?

File: libmobilecoin/Cargo.toml (hunk 1 above)
Comment (the-real-adammork) [reply libmobilecoin/Cargo.toml:17]: I was working off an older commit, but you're right it should be 2.1.0

File: libmobilecoin/include/transaction.h (hunk 2 above)
Comment (remoun) [reply libmobilecoin/include/transaction.h:62]: Skipping this, since there are no existing tests for this package.

**Review by remoun — APPROVED** [review]

**Pull request merged**